src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -78,18 +80,29 @@ private ExpressionStatementSyntax GenerateAssignment(ExpressionSyntax switchExpr
                     .WithLeadingTrivia(leadingTrivia);
             }
 
-            private StatementSyntax GenerateVariableDeclaration(ExpressionSyntax switchExpression, SyntaxTriviaList leadingTrivia)
+            private StatementSyntax GenerateVariableDeclaration(ExpressionSyntax switchExpression, SyntaxTriviaList leadingTrivia, ITypeSymbol variableSymbolTypeOpt)
             {
                 Debug.Assert(_assignmentTargetOpt is IdentifierNameSyntax);
 
-                return LocalDeclarationStatement(
+                if (variableSymbolTypeOpt != null && (variableSymbolTypeOpt.IsReferenceType || variableSymbolTypeOpt.IsNullable()))
```

Comment:
this needs to be commented.  i have no idea why this is the correct logic.  i.e. why are we only doing this for reference type and nullable value types?

Reply:
Might as well extract to a method with a self explanatory name.